x-pack/plugins/reporting/server/lib/esqueue/helpers/create_index.js

```diff
@@ -88,7 +88,17 @@ export function createIndex(client, indexName,
           index: indexName,
           body: body
         })
-          .then(() => true);
+          .then(() => true)
+          .catch(err => {
+            /* FIXME creating the index will fail if there were multiple jobs staged in parallel.
+             * Each staged job checks `client.indices.exists` and could each get `false` as a response.
+             * Only the first job in line can successfully create it though.
+             * The problem might only happen in automated tests, where the indices are deleted after each test run.
+             * This catch block is in place to not fail a job if the job runner hits this race condition.
+             * Unfortunately we don't have a logger in scope to log a warning.
+             */
+            err; // no-op
+          });
```

Comment:
Normally, we would want to log a warning here, but there is no logger in context in this function.

We could also check the `statusCode` on the err object to try to be certain it's a case of attempt to create an index that already exists. It isn't super specific though, the statusCode will just be 400.

Reply:
If creating the index fails, one way we might be able to catch that is to check for the index and its settings, which will include the mapping.

Reply:
Note: The FIXME was that we should log a warning here. That'll take a bit of refactoring to get a logger in scope